tests/NLog.UnitTests/LayoutRenderers/Log4JXmlTests.cs

```diff
@@ -333,5 +334,50 @@ void BadXmlValueTest()
             Assert.Contains("abc", badString);
             Assert.Contains("abc", goodString);
         }
+
+        [Fact]
+        public void NestedDiagnosticContextInformation_is_stored_in_the_same_sort_order_as_used_by_the_nested_diagnostic_layout_renderer()
```

Comment:
Don't think this test is relevant. As you have already asserted in the test above, that order is as expected.

Reply:
I aggree. Technically they are testing the same thing. 
But I think semantically, the second test makes the desired behavior more clear for other developers.

The old test already had forced the inverse sorting and had to be modified. Perhaps this assert should be removed from the general test? 

In addition I suggest adding another explicit test, which specifies that if both ndc and ndlc are present, they are combined with a `ndcSeparator` (as implemented).